btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/FeeService.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright D3 Ledger, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.d3.btc.withdrawal.service
+
+import com.d3.btc.helper.currency.satToBtc
+import com.d3.btc.withdrawal.transaction.WithdrawalDetails
+import com.d3.commons.sidechain.iroha.consumer.MultiSigIrohaConsumer
+import com.d3.commons.sidechain.iroha.util.ModelUtil
+import com.github.kittinunf.result.Result
+import com.github.kittinunf.result.flatMap
+import com.github.kittinunf.result.map
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Component
+import java.math.RoundingMode
+
+private const val BTC_ASSET_ID = "btc#bitcoin"
+// Key that signals that transfer is 'fee'
+const val FEE_KEY = "f842c3e1858df8c85794efacd06871f806afab04e0458c44488d2ffb3f071075"
+
+/**
+ * Service that is used to pay fees
+ */
+@Component
+class FeeService(
+    @Qualifier("withdrawalConsumerMultiSig")
+    private val withdrawalConsumer: MultiSigIrohaConsumer
+) {
+
+    /**
+     * Pays fee according to [withdrawalDetails]
+     * @param withdrawalDetails - details of withdrawal
+     * @return result of operation
+     */
+    fun payFee(withdrawalDetails: WithdrawalDetails): Result<Unit, Exception> {
+        val domain = withdrawalDetails.sourceAccountId.split("@")[1]
+        return withdrawalConsumer.getConsumerQuorum()
+            .flatMap { quorum ->
+                ModelUtil.transferAssetIroha(
+                    withdrawalConsumer,
+                    withdrawalConsumer.creator,
+                    "withdrawal_billing@$domain",
+                    BTC_ASSET_ID,
+                    "Fee",
+                    satToBtc(withdrawalDetails.withdrawalFeeSat).setScale(8, RoundingMode.DOWN).toPlainString(),
```

Comment:
`RoundingMode.UP` ?

Reply:
I think in this case user may be out of fee.